src/main/resources/META-INF/openapi.yaml

```diff
@@ -105,11 +105,6 @@ paths:
                     $ref: '#/components/schemas/AnalysisReport'
                   html_report:
                     type: object
-              encoding:
```

Comment:
If I am not wrong this is a valid optional field https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#encoding-object
And it documents the parts included in the multipart response

Reply:
I see, but the [openapi-generator](https://github.com/OpenAPITools/openapi-generator), more specifically the [maven plugin](https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator-maven-plugin) reports this as an error.
Users generating from this spec using this tool, can ignore this error by using the `skipValidateSpec` option. But this is not a good practice, skipping validation all together.

Maybe these keys are simply not in the write place?
Either that, or perhaps the documentation is wrong, or they have a bug in their validation tool.

Reply:
Actually, the validator for IntelliJ seems to be ok with that. So it may be a bug in the maven plugin eventually.
![Peek 2023-05-22 11-15](https://github.com/RHEcosystemAppEng/crda-backend/assets/28388442/2ad81e6e-f7b9-4549-9759-fc68d8885a3a)

Reply:
I would prefer to drop these properties regardless. Maybe temporarily until we figure this out. I don't want to use the _skipValidateSpec_, especially not this early in the project development stage.

But it's your call. Tell if you want to keep them, and I'll push a rollback.

Reply:
I prefer to split the API model from the backend project so that you can just reuse it from the cli and don't depend on the openapi generator tools.

Reply:
I don't understand.
So you don't want me to generate the backend types for the [crda-cli](https://github.com/RHEcosystemAppEng/crda-cli) and the [crda-java-api](https://github.com/RHEcosystemAppEng/crda-java-api) (and I assume also not for the future _crda-js-api_).

May I ask why? What is the reason for this request?

Because, not only will this result in large amounts of code duplicated between various projects.
But also, what are we using the backend's [openapi spec](https://github.com/RHEcosystemAppEng/crda-backend/blob/main/src/main/resources/META-INF/openapi.yaml) for if not for generating types for our various client side components?

Reply:
I'm only referring to the Java model. It will not duplicate code for Java. On the contrary.
Actually this crda-java-api that you just linked is what I was meaning about separating from the crda-backend project.

Reply:
I'm sorry, I not following what you're writing.
Copied from openapis.org:

```text
The OpenAPI Specification is a specification language for HTTP APIs that provides a standardized means to define your API to others.
You can quickly discover how an API works, configure infrastructure, generate client code, and create test cases for your APIs
```

This is exactly what we're doing, we created an [openapi spec](https://github.com/RHEcosystemAppEng/crda-backend/blob/main/src/main/resources/META-INF/openapi.yaml) for the backend. And we use it to generate (only) types for the [crda-java-api](https://github.com/RHEcosystemAppEng/crda-java-api) and [crda-cli](https://github.com/RHEcosystemAppEng/crda-cli) projects.

And you don't want us to do that? Why?
What is there to gain with here?

I honestly don't understand your point. Please reiterate it.

Reply:
If you don't want to remove the "encoded" directive because error is a "generator issue" that should be fixed in the maven plugin project. I get it, I'll skip verification until it's resolved.
But I do not understand why would you prefer I won't use the openapi spec at all.

Reply:
I'm following the code-first approach and from the code I manually update the OpenAPI spec (because the generated API when using Camel is not flexible enough)
That means that we have to maintain and update the model used in the backend. The same model you are generating again in this crda-java-api repository. I'm suggesting that, at least for Java, this API model is not generated because it already exists.

Reply:
I'm not generating the Java model.
Just the types for deserializing the backend response.

Reply:
> As for code-first approach, the code's spec is considered a part of the code in this approach. And if _Camel_ can't generate the spec automatically for you, and you don't want to do it manually. There are other tools you can introduce to the project for generating the spec, and can work a long side _camel_.

I prefer to do it manually because I have better control of all the spec details although sometimes I can make mistakes.

> But that's exactly what I think we should do. If you change the something in the return type for the backend, you should also update the spec in the same code-base. So your clients can catch up on that update and no one will be surprised.

OK, let's have a project for the openapi spec and generate the types for all the dependent languages in other separate projects as you already started doing. i.e. `crda-java-api`. Then in the backend I will add a dependency to this api.

Reply:
Agreed!